searchlib/src/tests/query/query-old.cpp

```diff
@@ -651,4 +651,80 @@ TEST("require that we do not break the stack on bad query") {
     EXPECT_FALSE(term.isValid());
 }
 
+TEST("testPhraseEvaluate") {
+    QueryBuilder<SimpleQueryNodeTypes> builder;
+    builder.addSameElement(3, "field");
+    {
+        builder.addStringTerm("a", "f1", 0, Weight(0));
+        builder.addStringTerm("b", "f2", 1, Weight(0));
+        builder.addStringTerm("c", "f3", 2, Weight(0));
+    }
+    Node::UP node = builder.build();
+    vespalib::string stackDump = StackDumpCreator::create(*node);
+    QueryNodeResultFactory empty;
+    Query q(empty, stackDump);
+    SameElementQueryNode * p = dynamic_cast<SameElementQueryNode *>(&q.getRoot());
+    EXPECT_TRUE(p != nullptr);
+    EXPECT_EQUAL("field", p->getIndex());
+    EXPECT_EQUAL(3u, p->size());
+    EXPECT_TRUE(dynamic_cast<const QueryTerm *>((*p)[0].get()) != nullptr);
```

Comment:
Consider helper function for the dynamic cast and name check

Reply:
Done